File: exam-api/src/main/java/com/camoleze/examapi/model/Exam.java
package com.camoleze.examapi.model;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.relational.core.mapping.Table;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("exams")
public class Exam {
    @Id
    private Long id;
    private String title;
    private String description;
    private Long hostUserId;
    private String joinCode;
    private ExamStatus status;
    private Integer timeLimit; // in minutes
    private Boolean allowRetake;
    
    @CreatedDate
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime createdAt;
    
    @LastModifiedDate
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime updatedAt;
    
    public enum ExamStatus {
        DRAFT, ACTIVE, COMPLETED, CANCELLED
    }
}

File: exam_app/lib/models/exam.dart
import 'package:json_annotation/json_annotation.dart';
import 'question.dart';

part 'exam.g.dart';

@JsonSerializable()
class Exam {
  final int id;
  final String title;
  final String? description;
  final int hostUserId;
  final String joinCode;
  final ExamStatus status;
  final int? timeLimit;
  final bool allowRetake;
  final DateTime? createdAt;
  final DateTime? updatedAt;
  final List<Question>? questions;

  Exam({
    required this.id,
    required this.title,
    this.description,
    required this.hostUserId,
    required this.joinCode,
    required this.status,
    this.timeLimit,
    required this.allowRetake,
    this.createdAt,
    this.updatedAt,
    this.questions,
  });

  factory Exam.fromJson(Map<String, dynamic> json) => _$ExamFromJson(json);
  Map<String, dynamic> toJson() => _$ExamToJson(this);
}

enum ExamStatus {
  @JsonValue('DRAFT')
  draft,
  @JsonValue('ACTIVE')
  active,
  @JsonValue('COMPLETED')
  completed,
  @JsonValue('CANCELLED')
  cancelled,
}